### PR TITLE
Made some changes and renamed files from Group8 to RotationInZ

### PR DESCRIPTION
**Pull request by ike-agbxje**

### CODE REVIEW

File: Group8/RotationInZ.c [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#include "RotationInZ.h"
```
Comment (andrewlinington) [inline Group8/RotationInZ.c:1]: Files names are correct but folder must be renamed too

File: groupIncludes.h [hunk 2]
```diff
@@ -16,7 +16,7 @@
 #include "ScalingInZ/ScalingInZ.h"
 #include "RotationInX/RotationInX.h"
 #include "RotationInY/RotationInY.h"
-#include "Group8/Group8.h"
+#include "Group8/RotationInZ.h"
```
Comment (andrewlinington) [inline groupIncludes.h:19]: Same as above

File: makefile [hunk 3]
```diff
@@ -2,7 +2,7 @@ CC = clang
...
+O_FILES = Group1/Group1.o GlobalScaling/GlobalScaling.o ScalingInX/ScalingInX.o ScalingInY/ScalingInY.o ScalingInZ/ScalingInZ.o RotationInX/RotationInX.o RotationInY/RotationInY.o Group8/RotationInZ.o XYZTranslation/XYZTranslation.o XPlaneReflection/XPlaneReflection.o YPlaneReflection/YPlaneReflection.o ZPlaneReflection/ZPlaneReflection.o ShearInX/ShearInX.o ShearInY/ShearInY.o ShearInZ/ShearInZ.o Group16/Group16.o OutputResults/OutputResults.o Group1/Group1Tests.o GlobalScaling/GlobalScalingTests.o ScalingInX/ScalingInXTests.o ScalingInY/ScalingInYTests.o ScalingInZ/ScalingInZTests.o RotationInX/RotationInXTests.o RotationInY/RotationInYTests.o Group8/RotationInZTests.o XYZTranslation/XYZTranslationTests.o XPlaneReflection/XPlaneReflectionTests.o YPlaneReflection/YPlaneReflectionTests.o ZPlaneReflection/ZPlaneReflectionTests.o ShearInX/ShearInXTests.o ShearInY/ShearInYTests.o ShearInZ/ShearInZTests.o Group16/Group16Tests.o OutputResults/OutputResultsTests.o
```
Comment (andrewlinington) [inline makefile:5]: Both group ones need fixed

**Review by andrewlinington — CHANGES_REQUESTED** [review]

**Review by andrewlinington — CHANGES_REQUESTED** [review]
image
some failing tests and an invalid free are present. fix these issues and it should be good to merge and if you like to have some code pushed up rright now comment out the tests and ill merge this as functionality seemed fine.

**Review by andrewlinington — APPROVED** [review]

**Review by zalgayya — APPROVED** [review]
looking good

**Pull request merged**